lib/rialto/etl/cli/publications.rb

```diff
@@ -1,27 +1,114 @@
 # frozen_string_literal: true
 
-require 'csv'
 require 'fileutils'
-require 'rialto/etl/cli/composite_etl'
+require 'jsonpath'
 
 module Rialto
   module Etl
     module CLI
-      # Publications iterates over the people in the input file CSV and for each
-      # calls extract, transform, and load.
-      class Publications < CompositeEtl
-        protected
+      # Publications grabs all Stanford University-affiliated publications from
+      # the WoS API, then transforms and loads them.
+      class Publications < Thor
+        option :input_directory,
+               default: 'data',
+               banner: 'INPUT',
+               desc: 'Name of directory with data to be transformed',
+               aliases: '-d'
+        option :output_directory,
+               default: 'data',
+               banner: 'OUTPUT',
+               desc: 'Name of directory to write transformed data to',
+               aliases: '-o'
+        option :force,
+               default: false,
+               type: :boolean,
+               banner: 'FORCE',
+               desc: 'Overwrite files that already exist',
+               aliases: '-f'
+        option :skip_extract,
+               default: false,
+               type: :boolean,
+               banner: 'SKIP_EXTRACT',
+               desc: 'Skip extract step'
+        option :skip_load,
+               default: false,
+               type: :boolean,
+               banner: 'SKIP_LOAD',
+               desc: 'Skip load step'
 
-        def file_prefix
-          'wos'
+        desc 'load', 'Extract, load, and transform for all Stanford University publications in WoS'
+        def load
+          FileUtils.mkdir_p(input_directory)
+
+          extract do |source_file|
+            sparql_file = transform(source_file)
+
+            load_sparql(sparql_file)
+          end
+        end
+
+        private
+
+        def cached_files(&_block)
+          Dir.glob("#{input_directory}/WOS*.json").each { |file_path| yield file_path }
```

Comment:
Are we going to run into a problem with too many files in directory to glob?

Reply:
Possibly? FWIW, POSIX bombed on `ls -l path/to/WOS*.json` with a limit error but the above code still worked. I'd guess the limit here is available memory rather than the POSIX default.